gazelle_haskell_modules/dependency_resolution.go

```diff
@@ -135,27 +136,44 @@ func findModuleLabelByModuleName(
 	libs []label.Label,
 ) (*label.Label, error) {
 	spec := moduleByModuleImportSpec(moduleImport)
+	isBootDep := false
+	// If the module we are considering comes from an `hs-boot` file,
+	// we store this in the `isBootDep` variable and then
+	// behave as if we were considering the standard version of ther module.
+	if strings.HasSuffix(spec.Imp, ".hs-boot") {
```

Comment:
The `strings.HasSuffix` check is used multiple times, it might be worth factoring out?